src/credentials.rs

```diff
@@ -65,24 +54,19 @@ impl Credentials {
     /// which schema is a valid `Basic` HTTP Authorization Schema.
     pub fn from_header(auth_header: String) -> Result<Credentials, AuthBasicError> {
         // check if its a valid basic auth header
-        let parts = auth_header.split(' ');
-        let parts: Vec<String> = parts.map(|part| part.to_string()).collect();
-
-        if parts.len() > 2 {
-            // invalid authorization token received
-            return Err(AuthBasicError::InvalidAuthorizationHeader);
-        }
+        if let Some((auth_type, encoded_credentials)) = auth_header.split_once(' ') {
+            if encoded_credentials.contains(' ') {
+                // Invalid authorization token received
+                return Err(AuthBasicError::InvalidAuthorizationHeader);
+            }
 
-        if let Some(auth_type) = parts.get(0) {
-            // check the provided authorization header
+            // Check the provided authorization header
             // to be a "Basic" authorization header
             if auth_type.to_lowercase() != "basic" {
-                return Err(AuthBasicError::InvalidScheme(auth_type.to_string()));
+                return Err(AuthBasicError::InvalidScheme(auth_type.into()));
             }
```

Comment:
Just wondering if there's any improvement on performance or any other gain by doing this change here?
I think using `to_string()` makes it more "explicit" as it doesn't rely on type inference and it reads exactly to what one's expect as a developer. `into()` relies on type inference, so you would have to check for the signature of the `InvalidSchema` variant to understand `into()` purpose here.

Just thinking from the developer perspective. What do you think?

Reply:
I just like `into()` more because it's shorter :D
I will change those back to `to_string()` as it's, like you said, more explicit.